float32 lower_angle
float32 upper_angle
float32 range
